### ISSUE TITLE
WITHOUT ROWID table support

### ISSUE THREAD

**Issue by John-He-928** [vitusortner/floor#358]
It would be great to support creating [WITHOUT ROWID tables](https://www.sqlite.org/withoutrowid.html) using the `@Entity` annotation like `@Entity(withoutRowid: true)`.

WITHOUT ROWID table annotation should be easy to implement by just appending `WITHOUT ROWID` in the CREATE TABLE statement. It may be useful in creating small tables with primary keys other than `INTEGER PRIMARY KEY`. One example would be Many-to-Many relationship table:

```sql
CREATE TABLE Gallery (
    id INTEGER PRIMARY KEY, 
    url TEXT
    /* some other fields */ );

CREATE TABLE Tag (
    id INTEGER PRIMARY KEY, 
    name TEXT
    /* some other fields */ );

CREATE TABLE GalleryTag (
    galleryId INTEGER REFERENCES Gallery,
    tagId INTEGER REFERENCES Tag,
    PRIMARY KEY(galleryId, tagId)
) WITHOUT ROWID;
```

**Comment by mqus** [comment]
I think this is a good feature to have and shouldn't take much effort to implement. `WITHOUT ROWID` does only work in sqlite but I think we don't have any plans to support other SQL dialects.

IMHO we can definitely accept pull requests for this.

**Comment by vitusortner** [comment]
Closed by #365
